src/main/java/com/diffplug/gradle/pde/ProductFileUtil.java

```diff
@@ -88,4 +95,20 @@ static void transformProductFile(StringPrinter printer, String line, PluginCatal
 			return Optional.empty();
 		}
 	}
+
+	static Properties extractProperties(String[] lines) {
+		Properties props = new Properties();
+		for (String line : lines) {
```

Comment:
Maybe `Map<String, String> props = new LinkedHashMap<>()` instead of Properties?